packages/candid/src/idl.ts

```diff
@@ -1315,6 +1323,14 @@ export class VariantClass extends ConstructType<Record<string, any>> {
     }
     throw new Error('Variant has no data: ' + x);
   }
+
+  get alternativesAsObject(): Record<string, Type> {
```

Comment:
Same again. Convenient, but maybe inefficient too? Also, shouldn't we be comparing the fields by their hashes, not their names? If there's a collision, something might go subtly wrong.

Reply:
Perf can come later